### ISSUE TITLE
WillCascadeOnDelete not working with EF Core

### ISSUE THREAD

**Issue by lizardking** [msawczyn/EFDesigner#22]
Hi
When I define relations between entities and set _End 1 on delete_ resp. _End 2 on delete_ to _Cascade_ or _None_ the EFDesigner creates a _.WillCascadeOnDelete(true/false)_. This did not compile on my system. AFIK _WillCascadeOnDelete_ does not exist for EF core (it does in EF6) and _OnDelete(DeleteBehavior)_ has to be used for EF Core.

Replacing the code to generate _WillCascadeOnDelete_ for the for undirectionalassociations with

```c#
switch(association.SourceDeleteAction) {
  case DeleteAction.Default:
    //Nothing to do
    break;
  case DeleteAction.Cascade:
    segments.Add("OnDelete(DeleteBehavior.Cascade)");
    break;
  case DeleteAction.None:
    segments.Add("OnDelete(DeleteBehavior.Restrict)");
    break;
}
```
and for the bidirectionalassociations with 
```c#
switch(association.SourceDeleteAction) {
  case DeleteAction.Default:
    //Nothing to do
    break;
  case DeleteAction.Cascade:
    segments.Add("OnDelete(DeleteBehavior.Cascade)");
    break;
  case DeleteAction.None:
    segments.Add("OnDelete(DeleteBehavior.Restrict)");
    break;
}
```
in the [EFCoreDesigner.ttinclude](https://github.com/msawczyn/EFDesigner/blob/master/src/DslPackage/TextTemplates/EFCoreDesigner.ttinclude) seems to fix the problem. Not sure if this would be valid for all use cases.

Small side note:
I'm not sure if I got this right, but for Bidirectionalassociations the docu on _End 1 on delete_ resp. _End 2 on delete_ says _How to handle objects on this end if the object on the other end is deleted_. However the T4 files for EF core, seem to handle this more like _How to handle objects on the other end if the object on this end is deleted_.

**Comment by msawczyn** [comment]
Thanks for the bug report. I'm looking into this; cascade delete support has always been a bit problematic and since you're bringing it up, it's a good time to review it again.

You've got a good start on the fix. There's a bit more to do, since really the flag deals with the principal and dependent ends of the relationship, and that's not exposed nicely in the designer. So I'll do a bit more with the design surface as well as the T4. 

If all goes right, this fix should be out in v1.2.6.7

**Comment by msawczyn** [comment]
The concept is to propagate deletion to the other end of the association depending on which end is deleted. So, if `a <--> b`, where `a` is `End1` and `b` is `End2` (which will be obvious from other data in the property editor), when cascade is requested on `End1`, deleting `a` also deletes `b`. If cascade is requested on `End2`, deleting `b` also deletes `a`.

While _technically_ we're deleting associations, that's not the common way to consider object deletion. Rather, it's just that ... object deletion, which is why there cascade behaviors are defined on the _classes_ that are at the ends of the associations. 

That being said, EF's cascade delete behavior is currently only defined for removing dependent objects, not the other way around. That's a blemish in the designer, and I'm planning to refine what's displayed in the property editor to make that clear, with parallel changes in code generation.

Good article re: cascade delete in EF6 (with some discussion of EFCore) at https://weblogs.asp.net/ricardoperes/entity-framework-pitfalls-cascade-deletes

Hope that makes sense.

**Comment by msawczyn** [comment]
Fixed in v1.2.6.7. Took what you offered and added a few more enhancements to the property editor so that it was clearer what was going to happen in the generated code. And I'll be adding more documentation on this -- it's currently extremely wimpy in this area -- as soon as I have a moment.

Thanks!
